bobtimus/src/lib.rs

```diff
@@ -315,16 +317,19 @@ mod tests {
         };
 
         let message1 = bob
-            .handle_create_swap(CreateSwapPayload {
-                alice_inputs: vec![AliceInput {
-                    outpoint: message0.input.previous_output,
-                    blinding_key: message0.input_blinding_sk,
-                }],
-                address_redeem: message0.address_redeem,
-                address_change: message0.address_change,
-                fee: message0.fee,
-                btc_amount: redeem_amount_bob,
-            })
+            .handle_create_swap(
+                serde_json::to_string(&CreateSwapPayload {
```

Comment:
Having to `to_string` this in the test is not ideal. Why did you choose to pass a string as the payload?

Reply:
It was what fixed the deserialisation error that we were encountering (caused by `SecretKey`'s deser implementation). I may be able to change this API and deserialize before we call it, so that we can pass in the payload struct instead.